### ISSUE TITLE
Integrated terminal: environment variables in launch path

### ISSUE THREAD

**Issue by ebugusey** [Microsoft/vscode#10810]
- VSCode Version: 1.4.0
- OS Version: Windows 7

Steps to Reproduce:

1. Install _Git for Windows_ in `C:\Users\<username>\AppData\Local\Programs\Git`.
2. Open some project.
3. Set `terminal.integrated.shell.windows` to `%LocalAppData%\\Programs\\Git\\bin\\bash.exe`.
4. Open integrated terminal.
5. Set `terminal.integrated.shell.windows` to `C:\\Users\\<username>\\AppData\\Local\\Programs\\Git\\bin\\bash.exe`.
6. Open integrated terminal.

What happens:

1. On 4: Got error: `The terminal process terminated with exit code: 1`
1. On 6: Integrated terminal opens as expected.

Expected:
It seems vscode doesn't recognize environment variables in integrated terminal paths. It should.

**Comment by Tyriar** [comment]
Duplicate #7344

**Comment by mihailik** [comment]
@Tyriar you've closed this as a duplicate of #7344 `terminal.integrated.shell.windows setting Doesn't support environment variables`.

But then <a href="https://github.com/Microsoft/vscode/issues/7344#issuecomment-279025081">you have closed</a> #7344 as a duplicate of 
#14973 `Set environment variable before starting a terminal`, which was (almost fully) implemented.

Creates a windy walk from issue to issue dropping relevance on the way :-)

In fact this is related to #2809 `Support environment variables when resolving values in settings`, which is not implemented at this point.

----

Hope that helps anyone trying to make sense of things. When #2809 is finally done, the fix for this issue would be:

Set `terminal.integrated.shell.windows` to `${env.LocalAppData}\\Programs\\Git\\bin\\bash.exe`
